libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -221,7 +222,11 @@ bool ExpressionCompiler::visit(Assignment const& _assignment)
 			// value lvalue_ref value lvalue_ref
 		}
 		m_currentLValue->retrieveValue(_assignment.location(), true);
-		appendOrdinaryBinaryOperatorCode(Token::AssignmentToBinaryOp(op), *_assignment.annotation().type);
+		if (Token::isShiftOp(target_op))
+			// shift only cares about the signedness of both sides
+			appendShiftOperatorCode(target_op, *_assignment.leftHandSide().annotation().type, *_assignment.rightHandSide().annotation().type);
```

Comment:
Is it ok to just use `type` here? What about number constants that need to be converted to their mobile type first?